src/interpolation/details/ArborX_InterpDetailsPolynomialBasis.hpp

```diff
@@ -0,0 +1,166 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_INTERP_DETAILS_POLYNOMIAL_BASIS_HPP
+#define ARBORX_INTERP_DETAILS_POLYNOMIAL_BASIS_HPP
+
+#include <Kokkos_Core.hpp>
+
+#include <type_traits>
+
+namespace ArborX::Interpolation
+{
+
+namespace Details
+{
+
+// Polynomial basis is computed in the same way as the following example
+// diagrams. Each cell is the product of the variable times everything on the
+// left cell and above.
+//
+// For example, the "degree 3 / variable y" has size 3 because it multiplies
+// every element of "degree 2 / variable y" and "degree 2 / variable x".
+//
+//   Quadratic |  3D  ||        Cubic      |  2D
+// ------------+----- || ------------------+-----
+//    degree   | var  ||       degree      | var
+// ---+---+----+      || ---+---+----+-----+
+//  0 | 1 |  2 |      ||  0 | 1 |  2 |  3  |
+// ---+---+----+      || ---+---+----+-----+
+// ---+---+----+      || ---+---+----+-----+
+//  1 |   |    |      ||  1 |   |    |     |
+// ---+---+----+----- || ---+---+----+-----+-----
+//    | x |    |  x   ||    | x |    |     |  x
+//    |   | xx |      ||    |   | xx |     |
+//    +---+----+----- ||    |   |    | xxx |
+//    | y |    |  y   ||    +---+----+-----+-----
+//    |   | xy |      ||    | y |    |     |  y
+//    |   | yy |      ||    |   | xy |     |
+//    +---+----+----- ||    |   | yy |     |
+//    | z |    |  z   ||    |   |    | xxy |
+//    |   | xz |      ||    |   |    | xyy |
+//    |   | yz |      ||    |   |    | yyy |
+//    |   | zz |      ||
+//
+// This generates:    || This generates:
+// [1, x, y, z, xx,   || [1, x, y, xx, xy, yy,
+//  xy, yy, xz, yz,   ||  xxx, xxy, xyy, yyy]
+//  zz]               ||
```

Comment:
I don't understand what you are trying to say.

Reply:
I have tried rewording it so it is clearer. However, I do not know if it is better to use diagrams, plain text or both.